src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -304,7 +304,28 @@ internal RuntimeType GetRuntimeType()
         }
 
         [MethodImpl(MethodImplOptions.InternalCall)]
-        internal static extern CorElementType GetCorElementType(RuntimeType type);
+        private static extern CorElementType GetCorElementTypeInternal(RuntimeType type);
+
+        internal static CorElementType GetCorElementType(RuntimeType type)
+        {
+            if (type is null)
```

Comment:
These null checks look unreachable. Nullability annotations 
guarantee that we use this on non-null types. I think it would be fine to change these null checks to asserts, or even better change these methods to instance methods on  RuntimeType. 

Also, as we are moving more bits and pieces to C#, we often find ourselves working with MethodTable* or the native TypeHandle. Should we change these FCalls to take the native TypeHandle so that one set can serve both reflection and non-reflection needs, and maybe also move them someplace where they can be discovered more easily?

It can look like this with both suggestions applied:
```
class RuntimeType
{
     internal CorElementType GetCorElementType()
     {
        CorElementType ret = GetNativeTypeHandle().GetCorElementType();
        GC.KeepAlive(this);
        return ret;
     }
}
```

Reply:
Yes, I will do this.